From 2c9e3ea090bb358d17029b6f973f84a654a6393e Mon Sep 17 00:00:00 2001
From: Neha Bisht <quic_nbisht@quicinc.com>
Date: Mon, 19 Sep 2022 16:50:49 +0530
Subject: [PATCH] skb_recycler: Add debug code to check if skb is recyclable.

Add debug code to check for various conditions on skb recyclability.

Change-Id: I073d90da38f9354cd7c15dcff64b404162d8b01c
Signed-off-by: Neha Bisht <quic_nbisht@quicinc.com>
---
 include/linux/skbuff.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index 99b9ad9f35ae..c1cbdafd83f5 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -1254,6 +1254,7 @@ static inline void consume_skb(struct sk_buff *skb)
 #endif
 
 void consume_skb_list_fast(struct sk_buff_head *skb_list);
+void check_skb_fast_recyclable(struct sk_buff *skb);
 void __consume_stateless_skb(struct sk_buff *skb);
 void  __kfree_skb(struct sk_buff *skb);
 extern struct kmem_cache *skbuff_head_cache;
@@ -1381,6 +1382,7 @@ static inline int skb_pad(struct sk_buff *skb, int pad)
 }
 #define dev_kfree_skb(a)	consume_skb(a)
 #define dev_kfree_skb_list_fast(a)	consume_skb_list_fast(a)
+#define dev_check_skb_fast_recyclable(a)       check_skb_fast_recyclable(a)
 
 int skb_append_pagefrags(struct sk_buff *skb, struct page *page,
 			 int offset, size_t size);
-- 
2.34.1

